### PR TITLE
build: bump to node 18.20 to support v18

### PR DESCRIPTION
**Pull request by JeanMeche**
This is mostly based on what @crisbeto did on #54412

### CODE REVIEW

File: tsconfig.json [hunk 1]
```diff
@@ -0,0 +1,9 @@
+{
```
Comment (alan-agius4) [inline tsconfig.json:1]: I don’t think this file is required.

Comment (JeanMeche) [reply tsconfig.json:1]: Without it, I've seen `yarn -s ng-dev pullapprove verify` fail with `TypeError: Cannot convert object to primitive value`.

Comment (alan-agius4) [reply tsconfig.json:1]: Umm. That looks a runtime error which shouldn’t be effected by the ts-node settings.

Comment (JeanMeche) [reply tsconfig.json:1]: My bad the actual error is more likely to be : 

```
Could not read configuration file at /Users/kyro/Developer/web/angular-projects/angular/.ng-dev/config.mjs.
```

`config.mjs` doesn't exist, it's a ts file: `config.mts`.

**Review by devversion — APPROVED** [review]

File: tsconfig.json [hunk 2]
```diff
@@ -0,0 +1,9 @@
...
+    // This tsconfig is required for our scripts that use `ts-node`.
+    // Due to https://github.com/TypeStrong/ts-node/issues/2094 we can't pass the parameters
+    // through the command line in Node 20+ so we have to do it through the tsconfig.
+    "extends": "./packages/tsconfig.json",
+    "ts-node": {
+        "transpileOnly": true
```
[the remark this answers is not in the document]
Comment (JeanMeche) [reply tsconfig.json:7]: It was mostly inspired by @crisbeto's work on  #54412 so maybe he has some more to add to that topic

[the remark this answers is not in the document]
Comment (JeanMeche) [reply tsconfig.json:7]: The benchmark scripts also rely on this.

[the remark this answers is not in the document]
Comment (devversion) [reply tsconfig.json:7]: does it break if you remove the config?

[the remark this answers is not in the document]
Comment (JeanMeche) [reply tsconfig.json:7]: Yes.

[the remark this answers is not in the document]
Comment (devversion) [reply tsconfig.json:7]: Not a big fan of another tsconfig personally. I would prefer:

- benchmark tsnode code uses the benchmark tsconfig if there is any
- ng-dev tsnode code uses the .ng-dev tsconfig

that would seem more correct and also saves us from an extra file

[the remark this answers is not in the document]
Comment (devversion) [reply tsconfig.json:7]: @JeanMeche is this possible?

[the remark this answers is not in the document]
Comment (JeanMeche) [reply tsconfig.json]: Sorry I missed you previous message, I'm looking at an alternative by droping `ts-node` in favor or `tsx`.

[the remark this answers is not in the document]
Comment (devversion) [reply tsconfig.json]: Sorry, was flying back last days (cross)— `tsx` looks neat. We should probably prepare for it in `angular/dev-infra`

File: package.json [hunk 3]
```diff
@@ -45,7 +45,7 @@
     "devtools:test": "bazelisk test --config snapshot-build --//devtools/projects/shell-browser/src:flag_browser=chrome -- //devtools/...",
     "docs": "[[ -n $CI ]] && echo 'Cannot run this yarn script on CI' && exit 1 || yarn bazel run //adev:serve --fast_adev",
     "docs:build": "[[ -n $CI ]] && echo 'Cannot run this yarn script on CI' && exit 1 || yarn bazel build //adev:build --fast_adev",
-    "benchmarks": "ts-node --esm scripts/benchmarks/index.mts"
+    "benchmarks": "tsx scripts/benchmarks/index.mts"
```
Comment (devversion) [inline package.json:48]:
```suggestion
    "benchmarks": "tsx --tsconfig=scripts/tsconfig.json scripts/benchmarks/index.mts"
```

**Review by devversion — APPROVED** [review]

**Review by devversion — APPROVED** [review]

**Comment by atscott** [comment]
This PR was merged into the repository by commit 5cf5f044607ad28ede76354397c6b1799e18da11.

**Review by RemedialGenius101 — APPROVED** [review]

**Pull request closed without merging**